xtask/src/main.rs

```diff
@@ -111,6 +111,7 @@ fn run_action(workspace: Workspace, target: &str, extra_args: &[String]) {
                 }
             } else {
                 cmd.env("YK_TRACING_KIND", tracing_kind);
+                cmd.env("CARGO_TARGET_DIR", "target_ext");
```

Comment:
I see what you are doing, but I'm not certain why it prevents stuff being rebuilt.

Reply:
By using a different target dir, this cargo invocation will no longer clobber the target dir used when compiling xtask with crates built using a different `RUSTFLAGS`.

Reply:
OK, let's add a comment to this effect.